app/customer.py

```diff
@@ -0,0 +1,55 @@
+import dataclasses
+import math
+from typing import Any
+
+
+@dataclasses.dataclass
+class Customer:
+    name: str
+    location: tuple
+    money: int
+    car: dict
+    product_cart: dict
+
+    @staticmethod
+    def fuel_to_shop(
+            fuel_cons: float, distance_to_shop: float, fuel_price: float
+    ) -> float:
+        return ((fuel_cons * distance_to_shop) / 100 * fuel_price) * 2
+
+    def dist_to_shop(self, shop_location: tuple) -> float:
+        return (
+            math.sqrt((shop_location[0] - self.location[0]) ** 2
+                      + (shop_location[1] - self.location[1]) ** 2)
+        )
+
+    def product_total(self, products: dict) -> list:
+        prod_total = sum(price * count for price, count in zip(
+            products.values(),
+            self.product_cart.values()
```

Comment:
As for me, it is wrong
because we do not always agree that products will be sorted by name the same as self.product_car
self.product_cart = {
    "milk": 4,
    "bread": 2,
    "butter": 5
}

products = {
    "milk": 2,
    "butter": 3.2,
    "bread": 1.5,
}
What your code output?
Correct output will be 27.0
Your code output 21.9

Reply:
look, my code fits readme and passes all the tests. If other order of variables, of course, the result will be wrong. If you want to get all possible outputs you should rewrite the tests. I refactored this code twice and spent 3 days on it. Sorry, but I have no additional time for rewriting this once again.

Reply:
Привіт, головна мета курсу - навчитись писати гарний код, а не просто проходити тести, бо здебільшого до таких моментів на роботі і більше всього докулупуються. Дуже важко зробити щоб тести покрили всі кейси, різні кейси приходять поки читаєш код і починаєш розуміти які саме умови не пройдуть цей уривок коду.
Якість коду може впливати на безпеку і ефективність будь якої програми. Крім того, чистий і легко читабельний код забезпечує зручність при обслуговуванні та розвитку програми в майбутньому.

Reply:
@avkpol ^^